HW2/P2/P2.py

```diff
@@ -74,9 +78,23 @@
     # You should explore different values for the number of locks in the medium
     # grained locking
     ########################################
-    N = 10
-    counts[:] = orig_counts
-    with Timer() as t:
-        move_data_medium_grained(counts, src, dest, 100, N)
-    assert counts.sum() == total, "Wrong total after move_data_medium_grained"
-    print("Medium grained correlated: {} seconds".format(t.interval))
+    T2 = []
+    for N in Nrange:
+        counts[:] = orig_counts
+        with Timer() as t:
+            move_data_medium_grained(counts, src, dest, 100, N)
+        assert counts.sum() == total, "Wrong total after move_data_medium_grained"
+        T2.append(t.interval)
+        print("Medium grained correlated N={}: {} seconds".format(N, t.interval))
+
+
+
+    plt.figure()
+    plt.plot(Nrange, T1, 'ro-', label="Uncorrelated")
+    plt.plot(Nrange, T2, 'bo-', label="Correlated")
+    plt.legend(loc="upper left")
+    plt.title("Run Time vs. Number of Elements Sharing a Lock")
+    plt.xlabel("N")
+    plt.ylabel("Time (s)")
+    plt.xticks(range(0,101,5))
+    plt.savefig('p2.png')
```

Comment:
Interesting plotting decision, I did bar graphs (can check them out here if you are interested: https://github.com/harvard-cs205/cs205-homework/pull/319/files) but perhaps your method is more clear visually.